src/Core/TypeReaders/GuildUserProxyTypeReader.cs

```diff
@@ -28,7 +31,13 @@ public override async Task<TypeReaderResult> ReadAsync(ICommandContext context,
             }
             else
             {
-                if (MentionUtils.TryParseUser(input, out ulong userId) || ulong.TryParse(input, out userId))
+                bool validSnowFlake = false;
+                ulong userId = 0;
+
+                if (ulong.TryParse(input, out userId)) 
+    	            validSnowFlake = SnowflakeUtils.FromSnowflake(userId).ToUnixTimeSeconds() > DISCORD_START_UNIX;
```

Comment:
In which cases could this be an invalid snowflake?

Reply:
If the resolved date of the snowflake is before the DISCORD_START_UNIX (January 1st 2015). This works because officially accounts were started being made later in 2015, as a result the first couple million (if not billion) values are not valid snowflakes.